aten/src/ATen/core/type.cpp

```diff
@@ -742,13 +749,28 @@ TupleTypePtr TupleType::createNamed(
     const c10::optional<c10::QualifiedName>& qualName,
     const std::vector<std::string>& field_names,
     const std::vector<TypePtr>& field_types) {
-  TORCH_INTERNAL_ASSERT(field_names.size() == field_types.size());
+      std::vector<std::pair<std::string, IValue>> fields;
+      for (size_t i = 0; i < field_names.size(); ++i) {
+        fields.emplace_back(std::pair<std::string, IValue>(field_names[i], IValue()));
+      }
+      return TupleType::createNamed(qualName, fields, field_types);
+    }
+
+
+TupleTypePtr TupleType::createNamed(
+    const c10::optional<c10::QualifiedName>& qualName,
+    const std::vector<std::pair<std::string, IValue>>& fields,
+    const std::vector<TypePtr>& field_types) {
+  TORCH_INTERNAL_ASSERT(fields.size() == field_types.size());
   std::vector<Argument> arguments;
-  for (size_t i = 0; i < field_names.size(); ++i) {
+  for (size_t i = 0; i < fields.size(); ++i) {
+    TORCH_CHECK(fields[i].second.tagKind() != "Tensor", "Tensors are "
```

Comment:
Consider also state the reason for not supporting Tensor type default value here.

Reply:
Fixed